### ISSUE TITLE
Variables resolver incorrectly reporting errors

### ISSUE THREAD

**Issue by darksinge** [serverless/serverless#9055]
Starting with version [2.26.0](https://github.com/serverless/serverless/releases/tag/v2.26.0), which includes a new parser and resolver implementation (coincidence?), I'm seeing errors when importing a secret from AWS Secrets Manager using the "ssm" syntax which includes the region the secret resides(i.e., `${ssm:/aws/reference/secretsmanager/MySecret~true}` is fine, but not `${ssm.us-east-1:/aws/reference/secretsmanager/MySecret~true}`).

The following works just fine in version 2.25.2 and lower:
```yaml
custom:
  my-secret: ${ssm.us-east-1:/aws/reference/secretsmanager/MySecret~true}

resources:
  Resources:
    DBCluster:
      Type: AWS::RDS::DBCluster
      Properties:
        MasterUsername: ${self:custom.my-secret.username}
        MasterUserPassword: ${self:custom.my-secret.password}
        ...
```

But starting in version 2.26.0 and up to 2.28.2, I get the following errors reported:
```yaml
Serverless: Deprecation warning: Variables resolver reports following resolution errors:
              - Cannot resolve variable at "resources.Resources.DBCluster.Properties.MasterUsername": Value not found at "self" source,
              - Cannot resolve variable at "resources.Resources.DBCluster.Properties.MasterUserPassword": Value not found at "self" source
            From a next major it we will be communicated with a thrown error.
            Set "variablesResolutionMode: 20210219" in your service config, to adapt to this behavior now
            More Info: https://www.serverless.com/framework/docs/deprecations/#NEW_VARIABLES_RESOLVER
```

Despite the deprecation warning, the minimal example I provided below ends up deploying to the desired state. With that in mind, I debated opening this as a bug due to the [link in the error message](https://www.serverless.com/framework/docs/deprecations/#NEW_VARIABLES_RESOLVER), which leads to this:

> It supports very same variable syntax, and is being updated with support for same resolution sources. Still as it has 
> improved internal resolution rules (which leave no room for ambiguity) in some edge cases it may report errors on which 
> old parser passed by.

I read that to mean the new parser may report *actual* errors which the old parser failed to catch. Since this ticket is about the new parser reporting errors on a valid `serverless.yml`, I decided to go ahead and submit this. Particularly because, If I used `variablesResolutionMode: 20210219` to have serverless throw an error instead of a deprecation warning, then the deployment would fail (when otherwise it would result in the desired state). Seems kind of misleading.

<details>
<summary><code>serverless.yml</code></summary>

```yaml
service: deleteme
frameworkVersion: '2'

provider:
  name: aws
  runtime: nodejs12.x
  lambdaHashingVersion: 20201221
  stage: ${opt:stage, 'v1'}
  region: ${opt:region, 'us-east-1'}

custom:
  rds:
    credentials: ${ssm.us-east-1:/aws/reference/secretsmanager/MySecret~true}

resources:
  Conditions:
    IsVirginia: !Equals [ '${self:provider.region}', 'us-east-1' ]

  Resources:
    MySecret:
      Type: AWS::SecretsManager::Secret
      Condition:
      - IsVirginia # Only create secret in us-east-1 so that read
      Properties:
        Name: MySecret
        GenerateSecretString:
          GenerateStringKey: password
          SecretStringTemplate: '{"username": "admin"}'
          ExcludeCharacters: !Join ['', ['/', '\', '@', "'", '"', '!', '$', ' ', '`']]

    SecretTargetAttachment:
      Type: AWS::SecretsManager::SecretTargetAttachment
      Condition:
      - IsVirginia
      Properties:
        SecretId: !Ref MySecret
        TargetId: !Ref DBCluster
        TargetType: AWS::RDS::DBCluster

    DBCluster:
      Type: AWS::RDS::DBCluster
      Properties:
        MasterUsername: ${self:custom.rds.credentials.username}
        MasterUserPassword: ${self:custom.rds.credentials.password}
        DatabaseName: mydb
        Engine: aurora-mysql
        EngineMode: provisioned
        DBSubnetGroupName: !Ref DBSubnetGroup
        VpcSecurityGroupIds:
        - xxxxx

    DBInstance:
      Type: AWS::RDS::DBInstance
      Properties:
        DBInstanceClass: db.t3.small
        DBClusterIdentifier: !Ref DBCluster
        DBSubnetGroupName: !Ref DBSubnetGroup
        Engine: aurora-mysql
        PubliclyAccessible: true

    DBSubnetGroup:
      Type: AWS::RDS::DBSubnetGroup
      Properties:
        DBSubnetGroupDescription: "My DB Subnet group"
        SubnetIds:
        - xxxxx
        - xxxxx
```

</details>

<details>
<summary><b><code>SLS_DEBUG=* npx serverless deploy</code> output</b></summary>

```
Serverless: Deprecation warning: Variables resolver reports following resolution errors:
              - Cannot resolve variable at "resources.Resources.DBCluster.Properties.MasterUsername": Value not found at "self" source,
              - Cannot resolve variable at "resources.Resources.DBCluster.Properties.MasterUserPassword": Value not found at "self" source
            From a next major it we will be communicated with a thrown error.
            Set "variablesResolutionMode: 20210219" in your service config, to adapt to this behavior now
            More Info: https://www.serverless.com/framework/docs/deprecations/#NEW_VARIABLES_RESOLVER
Serverless: Load command interactiveCli
Serverless: Load command config
Serverless: Load command config:credentials
Serverless: Load command config:tabcompletion
Serverless: Load command config:tabcompletion:install
Serverless: Load command config:tabcompletion:uninstall
Serverless: Load command create
Serverless: Load command install
Serverless: Load command package
Serverless: Load command deploy
Serverless: Load command deploy:function
Serverless: Load command deploy:list
Serverless: Load command deploy:list:functions
Serverless: Load command invoke
Serverless: Load command invoke:local
Serverless: Load command info
Serverless: Load command logs
Serverless: Load command metrics
Serverless: Load command print
Serverless: Load command remove
Serverless: Load command rollback
Serverless: Load command rollback:function
Serverless: Load command slstats
Serverless: Load command plugin
Serverless: Load command plugin
Serverless: Load command plugin:install
Serverless: Load command plugin
Serverless: Load command plugin:uninstall
Serverless: Load command plugin
Serverless: Load command plugin:list
Serverless: Load command plugin
Serverless: Load command plugin:search
Serverless: Load command config
Serverless: Load command config:credentials
Serverless: Load command rollback
Serverless: Load command rollback:function
Serverless: Load command upgrade
Serverless: Load command uninstall
Serverless: Load command login
Serverless: Load command logout
Serverless: Load command generate-event
Serverless: Load command test
Serverless: Load command dashboard
Serverless: Load command output
Serverless: Load command output:get
Serverless: Load command output:list
Serverless: Load command param
Serverless: Load command param:get
Serverless: Load command param:list
Serverless: Load command studio
Serverless: [AWS cloudformation 200 0.44s 0 retries] describeStacks({ StackName: 'regionbase' })
Serverless: [AWS ssm 200 0.475s 0 retries] getParameter({
  Name: '/aws/reference/secretsmanager/MySecret',
  WithDecryption: true
})
Serverless: [AWS cloudformation 200 0.333s 0 retries] describeStacks({ StackName: 'regionbase' })
Serverless: Invoke deploy
Serverless: Invoke package
Serverless: Invoke aws:common:validate
Serverless: Invoke aws:common:cleanupTempDir
Serverless: Packaging service...
Serverless: Invoke aws:package:finalize
Serverless: Invoke aws:common:moveArtifactsToPackage
Serverless: Invoke aws:common:validate
Serverless: Invoke aws:deploy:deploy
Serverless: [AWS cloudformation 200 0.326s 0 retries] describeStacks({ StackName: 'deleteme-v1' })
Serverless: [AWS cloudformation 200 0.325s 0 retries] describeStackResource({
  StackName: 'deleteme-v1',
  LogicalResourceId: 'ServerlessDeploymentBucket'
})
Serverless: [AWS s3 200 0.339s 0 retries] listObjectsV2({
  Bucket: 'deleteme-v1-serverlessdeploymentbucket-vpd21mnq5uv5',
  Prefix: 'serverless/deleteme/v1'
})
Serverless: [AWS s3 200 0.302s 0 retries] headObject({
  Bucket: 'deleteme-v1-serverlessdeploymentbucket-vpd21mnq5uv5',
  Key: 'serverless/deleteme/v1/1614726852167-2021-03-02T23:14:12.167Z/compiled-cloudformation-template.json'
})
Serverless: [AWS sts 200 0.31s 0 retries] getCallerIdentity({})
Serverless: Uploading CloudFormation file to S3...
Serverless: [AWS s3 200 0.395s 0 retries] putObject({
  Body: <Buffer 7b 22 41 57 53 54 65 6d 70 6c 61 74 65 46 6f 72 6d 61 74 56 65 72 73 69 6f 6e 22 3a 22 32 30 31 30 2d 30 39 2d 30 39 22 2c 22 44 65 73 63 72 69 70 74 ... 2135 more bytes>,
  Bucket: 'deleteme-v1-serverlessdeploymentbucket-vpd21mnq5uv5',
  Key: 'serverless/deleteme/v1/1614728087026-2021-03-02T23:34:47.026Z/compiled-cloudformation-template.json',
  ContentType: 'application/json',
  Metadata: { filesha256: 'xxxxxxxxxxxx' }
})
Serverless: Uploading artifacts...
Serverless: Validating template...
Serverless: [AWS cloudformation 200 0.502s 0 retries] validateTemplate({
  TemplateURL: 'https://s3.amazonaws.com/deleteme-v1-serverlessdeploymentbucket-vpd21mnq5uv5/serverless/deleteme/v1/1614728087026-2021-03-02T23:34:47.026Z/compiled-cloudformation-template.json'
})
Serverless: Updating Stack...
Serverless: [AWS cloudformation 200 0.94s 0 retries] updateStack({
  StackName: 'deleteme-v1',
  Capabilities: [ 'CAPABILITY_IAM', 'CAPABILITY_NAMED_IAM', [length]: 2 ],
  Parameters: [ [length]: 0 ],
  TemplateURL: 'https://s3.amazonaws.com/deleteme-v1-serverlessdeploymentbucket-vpd21mnq5uv5/serverless/deleteme/v1/1614728087026-2021-03-02T23:34:47.026Z/compiled-cloudformation-template.json',
  Tags: [ { Key: 'STAGE', Value: 'v1' }, [length]: 1 ]
})
Serverless: Checking Stack update progress...
Serverless: [AWS cloudformation 200 0.754s 0 retries] describeStackEvents({
  StackName: 'arn:aws:cloudformation:us-east-1:xxxxxxxxxxxx:stack/deleteme-v1/a9027100-7ba2-11eb-bcf4-0a4194e8a939'
})
...
.Serverless: [AWS cloudformation 200 0.686s 0 retries] describeStackEvents({
  StackName: 'arn:aws:cloudformation:us-east-1:xxxxxxxxxxxx:stack/deleteme-v1/a9027100-7ba2-11eb-bcf4-0a4194e8a939'
})
....
Serverless: Stack update finished...
Serverless: Invoke aws:info
Serverless: [AWS cloudformation 200 0.319s 0 retries] describeStacks({ StackName: 'deleteme-v1' })
Serverless: [AWS cloudformation 200 0.35s 0 retries] listStackResources({ StackName: 'deleteme-v1', NextToken: undefined })
Service Information
service: deleteme
stage: v1
region: us-east-1
stack: deleteme-v1
resources: 7
api keys:
  None
endpoints:
  None
functions:
  None
layers:
  None
Serverless: Invoke aws:deploy:finalize
Serverless: [AWS s3 200 0.337s 0 retries] listObjectsV2({
  Bucket: 'deleteme-v1-serverlessdeploymentbucket-vpd21mnq5uv5',
  Prefix: 'serverless/deleteme/v1'
})
Serverless: Removing old service artifacts from S3...
Serverless: [AWS s3 200 0.346s 0 retries] deleteObjects({
  Bucket: 'deleteme-v1-serverlessdeploymentbucket-vpd21mnq5uv5',
  Delete: {
    Objects: [
      {
        Key: 'serverless/deleteme/v1/1614726362687-2021-03-02T23:06:02.687Z/compiled-cloudformation-template.json'
      },
      {
        Key: 'serverless/deleteme/v1/1614726362687-2021-03-02T23:06:02.687Z/deleteme.zip'
      },
      [length]: 2
    ]
  }
})
```

</details>

<b>Installed version</b>

```
Framework Core: 2.28.2 (local)
Plugin: 4.4.3
SDK: 2.3.2
Components: 3.7.1
```

**Comment by medikoo** [comment]
@darksinge Thanks for report, still I believe that was fixed with v2.28.3. Please confirm

**Comment by darksinge** [comment]
@medikoo Yes, this has been fixed, thanks!
